rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/LoggersWindow.java

```diff
@@ -0,0 +1,207 @@
+        loggerField.setRequired(true);
+
+        ListGridField levelField = new ListGridField("level", "Level");
+        levelField.setValueMap("TRACE", "DEBUG", "INFO", "WARN", "ERROR", "FATAL", "OFF");
+        levelField.setCanEdit(true);
+
+        ListGridField removeField = new ListGridField(" ", 20);
+        removeField.setType(ListGridFieldType.IMAGE);
+        removeField.setCanEdit(false);
+        removeField.setCanSort(false);
+        removeField.setDefaultValue("blank.png");
+
+        loggersGrid.setFields(new ListGridField[] { loggerField, levelField, removeField });
+        loggersGrid.setCanResizeFields(true);
+        loggersGrid.setValidateOnChange(true);
+
+        this.controller.fetchLoggersSettings(loggersGrid);
+
+        removeField.addRecordClickHandler(event -> {
+            if (loggersGrid.getRecord(event.getRecordNum())
+                           .getAttribute(" ")
+                           .equals(Images.instance.remove().getSafeUri().asString())) {
+                loggersGrid.removeData(loggersGrid.getRecord(event.getRecordNum()));
+            }
+        });
+
+        IButton addRecordButton = new IButton("Add");
+        addRecordButton.setTooltip("Add a new logger record");
+        addRecordButton.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent event) {
+                ListGridRecord newRecord = new ListGridRecord();
+                loggersGrid.addData(newRecord);
+                newRecord.setAttribute("level", "INFO");
+                newRecord.setAttribute(" ", Images.instance.remove().getSafeUri().asString());
+                newRecord.set_canEdit(true);
+                loggersGrid.startEditing(loggersGrid.getRowNum(newRecord));
+                newRecord.setAttribute("logger", "");
+                newRecord.set_canRemove(true);
+            }
+        });
+
+        IButton resetGridButton = new IButton("Reset");
+        resetGridButton.setTooltip("Reset loggers settings");
+        resetGridButton.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent event) {
+                controller.fetchLoggersSettings(loggersGrid);
+                loggersGrid.redraw();
+            }
+        });
+
+        IButton applyChangesButton = new IButton("Apply");
+        applyChangesButton.setTooltip("Apply changes");
+        applyChangesButton.setIcon(Images.instance.ok_16().getSafeUri().asString());
+        applyChangesButton.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent event) {
+                loggersGrid.endEditing();
+                if (loggersGrid.hasErrors()) {
+                    for (int j = 0; j < loggersGrid.getTotalRows(); j++) {
+                        if (!loggersGrid.validateRow(j)) {
+                            loggersGrid.startEditing(j);
+                            return;
+                        }
+                    }
+                } else {
```

Comment:
How about doing `if (loggersGrid.hasChanges())` to avoid doing save/refresh process unnecessarily ?

Reply:
I tried it and did not work, I leave it like this !!